src/platform_impl/linux/drm/event_loop.rs

```diff
@@ -0,0 +1,1417 @@
+                        let dev_seat_name = device
+                            .property_value("ID_SEAT")
+                            .map(|x| x.to_os_string())
+                            .unwrap_or_else(|| std::ffi::OsString::from("seat0"));
+                        if dev_seat_name == seat_name {
+                            if let Ok(Some(pci)) = device.parent_with_subsystem(Path::new("pci")) {
+                                if let Some(id) = pci.attribute_value("boot_vga") {
+                                    return id == "1";
+                                }
+                            }
+                        }
+                        false
+                    })
+                    .flat_map(|device| device.devnode().map(std::path::PathBuf::from))
+                    .next()
+                    .or_else(|| {
+                        enumerator
+                            .scan_devices()
+                            .ok()?
+                            .filter(|device| {
+                                device
+                                    .property_value("ID_SEAT")
+                                    .map(|x| x.to_os_string())
+                                    .unwrap_or_else(|| std::ffi::OsString::from("seat0"))
+                                    == seat_name
+                            })
+                            .flat_map(|device| device.devnode().map(std::path::PathBuf::from))
+                            .next()
+                    })
+                    .ok_or(crate::error::OsError::new(
+                        line!(),
+                        file!(),
+                        crate::platform_impl::OsError::DrmMisc("Failed to find suitable GPU"),
+                    ))
+            },
+            |p| Ok(Into::into(p)),
+        )?;
+        let dev = seat.open_device(&card_path).map_err(|_| {
+            crate::error::OsError::new(
+                line!(),
+                file!(),
+                crate::platform_impl::OsError::DrmMisc("Failed to initialize DRM"),
+            )
+        })?;
+        let drm = Card(std::sync::Arc::new(dev.1));
+        let mut input = input::Libinput::new_with_udev(Interface(seat, HashMap::new()));
+        input.udev_assign_seat(seat_name).unwrap();
+        let xkb_ctx = xkb::Context::new(xkb::CONTEXT_NO_FLAGS);
+        let keymap = xkb::Keymap::new_from_names(
+            &xkb_ctx,
+            "",
+            "",
+            "",
+            "",
+            std::env::var("WINIT_XKB_OPTIONS").ok(),
+            xkb::KEYMAP_COMPILE_NO_FLAGS,
+        )
+        .ok_or(crate::error::OsError::new(
+            line!(),
+            file!(),
+            crate::platform_impl::OsError::DrmMisc("Failed to compile XKB keymap"),
+        ))?;
+        let state = xkb::State::new(&keymap);
+        let compose_table = xkb::compose::Table::new_from_locale(
+            &xkb_ctx,
+            std::env::var_os("LC_ALL")
```

Comment:
The code is very dense and hard to read, would suggest to logically space it, at least by operation kind. This applies to everything from what I can say in that PR.

Reply:
Fixed